src/lookups/lookup.ts

```diff
@@ -0,0 +1,99 @@
+import {Component, ChangeDetectionStrategy, Input, Attribute, Output, EventEmitter, ElementRef, Renderer, ChangeDetectorRef} from 'angular2/core';
+import {Control} from 'angular2/common';
+import 'rxjs/Rx';
+import {Observable} from 'rxjs/Observable';
+
+@Component({
+  selector: 'ngl-lookup',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl: './lookup.jade',
+})
+export class NglLookup {
+
+  @Input() placeholder: string;
+
+  @Input() set value(value: string) {
+    if (value !== this.lookup.value) {
+      this.lookup.updateValue(value);
+    }
+  }
+
+  @Input() filter: Function;
```

Comment:
This is probably a matter of personal taste but if the input was called `lookup` instead of `filter`, it's name would be closer to it's functionality, I believe.